### PR TITLE
Fix: wrong defaults and defaults not matching stated defaults

### PR DESCRIPTION
**Pull request by davhdavh**
<!--  Thanks for sending a pull request!  **What type of PR is this?**

/kind bug
/kind documentation

**What this PR does / why we need it**:

defaults for csi proxy used wrong image. And was missing username in actual config (only had the defaults in the docs)

**Requirements**:
- [x] uses [conventional commit messages]
- [x] includes documentation
- [x] tested upgrade from previous version

**Release note**:
```
Fix: CSI proxy for windows defaults did not match documented values
Fix: CSI proxy for windows defaults used wrong image
```

### CODE REVIEW

**Comment by davhdavh** [comment]
Err, where did my description disappear off to?!?

/kind bug
/kind documentation

**Release note**:
```
Fix: CSI proxy for windows defaults did not match documented defaults
Fix: CSI proxy for windows defaults used wrong image
```

File: charts/latest/csi-driver-smb/values.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ image:
     tag: v2.8.0
     pullPolicy: IfNotPresent
   csiproxy:
-    repository: ghcr.io/kubernetes-sigs/sig-windows/csi-provisioner
+    repository: ghcr.io/kubernetes-sigs/sig-windows/csi-proxy
```
Comment (andyzhangx) [inline charts/latest/csi-driver-smb/values.yaml:20]: what's this change?

File: charts/latest/csi-driver-smb/values.yaml [hunk 2]
```diff
@@ -147,6 +147,7 @@ windows:
     dsName: csi-proxy-win # daemonset name
     tolerations: {}
     affinity: {}
+    username: "NT AUTHORITY\\SYSTEM"
```
Comment (andyzhangx) [inline charts/latest/csi-driver-smb/values.yaml:150]: what's this change?

File: charts/latest/csi-driver-smb/values.yaml (hunk 1 above)
Comment (davhdavh) [reply charts/latest/csi-driver-smb/values.yaml:20]: Fix: CSI proxy for windows defaults used wrong image

File: charts/latest/csi-driver-smb/values.yaml (hunk 2 above)
Comment (davhdavh) [reply charts/latest/csi-driver-smb/values.yaml:150]: Fix: CSI proxy for windows defaults did not match documented defaults

File: charts/latest/csi-driver-smb/values.yaml (hunk 1 above)
Comment (andyzhangx) [reply charts/latest/csi-driver-smb/values.yaml:20]: `ghcr.io/kubernetes-sigs/sig-windows/csi-proxy:v1.1.2` does not exist

```
# docker pull ghcr.io/kubernetes-sigs/sig-windows/csi-proxy:v1.1.2
v1.1.2: Pulling from kubernetes-sigs/sig-windows/csi-proxy
no matching manifest for linux/amd64 in the manifest list entries
```

Comment (davhdavh) [reply charts/latest/csi-driver-smb/values.yaml:20]: ofcourse not, it is a windows image

**Comment by andyzhangx** [comment]
could you rebase this PR? thanks. @davhdavh

**Comment by davhdavh** [comment]
I have no idea how. git rebase ends up in a state where "You have divergent branches and need to specify how to reconcile them."

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**